src/coreclr/jit/gentree.cpp

```diff
@@ -19791,6 +19791,50 @@ uint16_t GenTreeLclVarCommon::GetLclOffs() const
     }
 }
 
+#if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
+//------------------------------------------------------------------------
+// GetOverwrittenOpNumForFMA: check if the result is written into one of the operands.
+// In the case that none of the operand is overwritten, check if any of them is lastUse.
+//
+// Return Value:
+//     The operand number overwritten or lastUse. 2 is the default value, where no overwritten and no lastUse.
+//
+unsigned GenTreeHWIntrinsic::GetOverwrittenOpNumForFMA(GenTree* use, GenTree* op1, GenTree* op2, GenTree* op3)
+{
+    // only FMA intrinsic node should call into this function
+    assert(HWIntrinsicInfo::lookupIsa(gtHWIntrinsicId) == InstructionSet_FMA);
+    unsigned overwrittenOpNum = 2; // 1->op1, 2->op2, 3->op3
```

Comment:
Why `2` instead of `1` (or simply `0` indicating none found)?

In particular, if we do `2` won't that mean we try to swap `op1`/`op2` in the default case rather than simply preserving it exactly as given?

Reply:
The default case happens when there is no overwritten and no lastUse. ```2``` was chosen because I thought in this default case, the fma form does not make any difference.

But after thinking about it, I think you are correct . Because in default case, the fma form should still be decided by which op ```isContained```. Merge default case with ```2``` would miss the possibility that ```op2``` is contained. We should still have a separate default case (0) and check all the possibilities.

Reply:
@tannergooding, do you have more comments besides these two questions? I will update the default case today. Does the code logic look correct to you?

Reply:
Yes, the rest of the logic looked correct to me.